osafw-app/App_Code/fw/ConvUtils.cs

```diff
@@ -236,4 +238,42 @@ private static string _replace_specials(string html_data)
         html_data = html_data.Replace(((char)(174)).ToString(), "<sup><small>R</small></sup>");
         return html_data;
     }
+
+    /// <summary>
+    /// To make this worh generic class should gave get and set for every property that expected to be converted.
+    /// 
+    /// Example:
+    /// class MyClass
+    /// {
+    ///    public int id { get; set; }
+    ///    public string name { get; set; }
+    /// }
+    /// Hashtable ht = new Hashtable() { { "id", 1 }, { "name", "John" } };
+    /// MyClass myClass = new MyClass();
+    /// myClass = ConvUtils.HashtableToClass<MyClass>(ht);
+    /// 
+    /// </summary>
+    /// <typeparam name="T">Class that hastable be converted to </typeparam>
+    /// <param name="hashtable">One level deep hashtable where keys has the same name as class properties</param>
+    /// <returns>class of passed generic type</returns>
+    public static T DictionaryToClass<T>(Dictionary<string, string> hashtable) where T : new()
```

Comment:
It might also worth to have a loop on Type properties like:

```
// Loop over *public* properties of T
    foreach (var prop in typeof(T).GetProperties())
    {
        if (!prop.CanWrite) continue; // skip read-only properties

        // property name might match the DB field exactly ("id", "iname", etc.)
        string propName = prop.Name;

        if (!dbrow.ContainsKey(propName)) 
              continue;
```

Maybe it's worth to test what's better for performance.


Also... in case of array calls when we have a lot of rows - need some sort of caching as reflection is not superfast. Maybe storing compiled delegates?